R/quosure-class.R

```diff
@@ -0,0 +1,29 @@
+#' Reproducible class with environment and code.
+#'
+#' Reproducible class with environment and code.
+#' @name Quosure-class
+#' @rdname Quosure-class
+#' @slot code (`character`) to reproduce the environment
+#' @slot env (`environment`) environment which content was generated by the evaluation
+#'  of the `code` slot.
+#' @slot id (`integer`) random identifier of the code element to make sure uniqueness
+#'  when joining.
+#' @keywords internal
+setClass(
+  "Quosure",
+  slots = c(env = "environment", code = "character", id = "integer"),
```

Comment:
ID should be an attribute of each element of `code` vector. This would save you assertions of the length between those two and possibly more.

Reply:
we can do this but please know that you can't assign attribute to a single element of the expression because it's like a vector.


```r
a <- as.expression(quote(a <- 1))
attr(a, "id") <- sample.int(.Machine$integer.max, size = 1)

b <- as.expression(quote(b <- 2)) 
attr(b, "id") <- sample.int(.Machine$integer.max, size = 1)

code <- c(a, b)
attr(code[1], "id")
attr(code[2], "id")
```

we will need to do something like this:

```r
code <- as.expression(quote(a <- 1))
attr(code, "id") <- sample.int(.Machine$integer.max, size = 1)

new_code <- as.expression(quote(b <- 2)) 
new_attrs <- c(attr(code, "id"), sample.int(.Machine$integer.max, size = 1))
code <- c(code, new_code)
attr(code, "id") <- new_attrs
```

If you feel that this is important please create the issue in teal.code

Reply:
How about the following?
```
a <- as.expression(quote(a <- 1))
attr(a, "id") <- sample.int(.Machine$integer.max, size = 1)

b <- as.expression(quote(b <- 2)) 
attr(b, "id") <- sample.int(.Machine$integer.max, size = 1)

code <- list(a, b)
attr(code[[1]], "id")
attr(code[[2]], "id")
```

or alternatively:

```
a <- setNames(list(as.expression(quote(a <- 1))), sample.int(.Machine$integer.max, size = 1))
b <- setNames(list(as.expression(quote(b <- 2))), sample.int(.Machine$integer.max, size = 1))
c(a, b)
```

Reply:
1. is not enough because you can provide code as n-element expression 
```r
eval_code(<qenv>, as.expression(quote(a <- 1), quote(b <- 2)))
```
There will be some additional code to loop through each and change from vector to a list

2. Look how funny
```r
a <- c("a", "b", "c")
names(a) <- c(3, 2, 1)
a[1]
#  3 
# "a" 
```

Reply:
new issue: https://github.com/insightsengineering/teal.code/issues/70